packages/api/src/routes/batchRoutes.js

```diff
@@ -1,12 +1,33 @@
+import isAfter from 'date-fns/is_after';
+import addDays from 'date-fns/add_days';
 import { Router } from 'express';
 import { asyncHandler } from '../services/asyncHandler';
-import { getAllBatches } from '../db/collections/batch';
+import { getAllBatches, insertBatch } from '../db/collections/batch';
 
 const batchRoutes = Router();
 
-batchRoutes.post('/list', asyncHandler(async (req, res, next) => {
+batchRoutes.get('/list', asyncHandler(async (req, res, next) => {
   const batchList = await getAllBatches();
-  res.json({ batchList });
+  const { filter } = req.query;
+  if (filter === 'active') {
+    const activeBatches = batchList.filter((batch) => {
+      const date = new Date(batch.endDate);
+      return isAfter(date, Date.now());
+    });
+    res.json({ activeBatches });
+  } else {
+    res.json({ batchList });
+  }
+  next();
```

Comment:
Since the purpose of this route has been fulfilled by the time we reach this line, we don't really need to call next here.